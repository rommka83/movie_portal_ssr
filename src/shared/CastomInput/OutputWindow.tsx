import React from 'react';
import styles from './outputWindow.module.css';
import classNames from 'classnames';

interface ICastomInputProps extends React.ComponentPropsWithoutRef<'div'> {
  name?: string;
  value?: string | number | null;
}

export function OutputWindow({ className, name, value }: ICastomInputProps) {
  return value && value !== '' ? (
    <div className={classNames(styles.label, className)}>
      {name && <p className={styles.labelText}>{name}:</p>}
      <p className={styles.content}>{value}</p>
    </div>
  ) : null;
}
