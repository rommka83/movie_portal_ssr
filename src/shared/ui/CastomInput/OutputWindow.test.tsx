import renderer from 'react-test-renderer';
import { OutputWindow } from './OutputWindow';

describe('проверка OutputWindow', () => {
  test('отрисовка OutputWindow с фотографией', () => {
    const component = renderer.create(<OutputWindow />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
