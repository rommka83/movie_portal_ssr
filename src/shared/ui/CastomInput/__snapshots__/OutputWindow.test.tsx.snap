// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка OutputWindow отрисовка OutputWindow с фотографией 1`] = `null`;
